### PR TITLE
[flang][NVPTX] Add initial support to the NVPTX target

### PR DESCRIPTION
**Pull request by fabianmcg**
This patch adds initial support to the NVPTX target, enabling `flang` to produce OpenMP offload code for NVPTX targets.

### CODE REVIEW

**Comment by fabianmcg** [comment]
Ping for review.

**Review by kiranchandramohan — APPROVED** [review]
Thanks for this contribution. This LGTM.

Have you tested the offloading tests on an Nvidia GPU?

File: flang/lib/Frontend/FrontendActions.cpp [hunk 1]
```diff
@@ -175,6 +175,45 @@ getExplicitAndImplicitAMDGPUTargetFeatures(CompilerInstance &ci,
   return llvm::join(featuresVec, ",");
 }
 
+// Get feature string which represents combined explicit target features
+// for NVPTX and the target features specified by the user/
+// TODO: Have a more robust target conf like `clang/lib/Basic/Targets/NVPTX.cpp`
```
Comment (kiranchandramohan) [inline flang/lib/Frontend/FrontendActions.cpp:180]: If this code can be refactored and placed in `llvm/lib/Frontend/Driver` then both clang and flang can share it.

**Comment by fabianmcg** [comment]
> Have you tested the offloading tests on an Nvidia GPU?

Yes, on an NVIDIA V100.

File: flang/lib/Frontend/FrontendActions.cpp (hunk 1 above)
Comment (fabianmcg) [reply flang/lib/Frontend/FrontendActions.cpp:180]: Sure, it should be doable, `AMDGPU` also needs a similar refactor, however, I'm thinking (and waiting) in adding them to `llvm/offload` as MLIR would also benefit from these refactors.

**Review by TIFitis — APPROVED** [review]
Looks good!

**Pull request merged**